src/services/document-service.js

```diff
@@ -291,17 +291,22 @@ class DocumentService {
     }
 
     const latestRevision = await this.getCurrentDocumentRevisionByKey(documentKey);
-    const order = await this.documentOrderStore.getNextOrder();
 
-    const newRevision = this._buildDocumentRevision({ data: latestRevision, documentKey, userId: user._id, order, sections: latestRevision.sections });
-
-    newRevision.appendTo = {
-      key: documentKey,
-      ancestorId: latestRevision._id
+    const doc = {
+      title: latestRevision.title,
+      slug: latestRevision.slug,
+      namespace: latestRevision.namespace,
+      language: latestRevision.language,
+      sections: latestRevision.sections,
+      appendTo: {
+        key: documentKey,
+        ancestorId: latestRevision._id
+      },
+      tags: latestRevision.tags,
+      archived
```

Comment:
`getNextOrder` is called by `createNewDocumentRevision` (as is `_buildDocumentRevision`). Also the check if it is an internal doc is done in `createNewDocumentRevision`, so all that remains to be done here is to set the `archived` prop on the last revision and save it again.

Reply:
The reason for `_buildDocumentRevision` was that it unifies revision creation. What I dislike in composing it in 2 places is that it leaves room for error and inconsistency.

Reply:
It's not really composing in 2 places, I would say. `_buildDocumentRevision` builds a data structure that is not really the one expected by `createNewDocumentRevision` (it puts much more properties onto the "doc"/"data" object). Even though that doesn't hurt in theory, I disliked calling `getNextOrder` twice. I don't know how to solve this, except maybe validating the params of `createNewDocumentRevision` with `joi`, so we can be sure the data is 100% correct.

Reply:
Ok then maybe let's do that then, the joi validation in a separate ticket 🤔

Reply:
Great idea!